### PR TITLE
Apply patch

### PR DESCRIPTION
**Pull request by spyrkob**
Part of #42 - apply-patch operation

Needs some more validation, but the PR is getting big enough, so I'll add it later.

@TomasHofman could you review when you have a moment?

### CODE REVIEW

**Comment by TomasHofman** [comment]
Sure - I will take a bit more time on this one, will try to do it later today.

File: prospero-common/src/main/java/org/wildfly/prospero/actions/ApplyPatch.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+        GalleonUtils.executeGalleon(options -> provMgr.provision(provMgr.getProvisioningConfig(), options),
+                mavenSessionManager.getProvisioningRepo().toAbsolutePath());
+
+        metadata.setChannel(maven.resolvedChannel());
+
+        metadata.writeFiles();
+    }
+
+    private List<Channel> mapToChannels(List<ChannelRef> channelRefs) throws MetadataException {
```
Comment (TomasHofman) [inline prospero-common/src/main/java/org/wildfly/prospero/actions/ApplyPatch.java:118]: Just an idea, we could create a conversion method on the ChannelRef class, e.g. `ChannelRef#toChannel()` (non-static), and then do these conversions via streams, like `channelRefs.stream().map(ChannelRef::toChannel()).collect(...)`. Or maybe have some static utility methods to convert a single instance or a whole collection. It's just that I think this method already exists in a different class. Can be a future task...

**Review by TomasHofman — APPROVED** [review]
I think let's merge.

Would the patch repo remain in the .installation/ permanently?

File: prospero-common/src/main/java/org/wildfly/prospero/actions/ApplyPatch.java (hunk 1 above)
Comment (spyrkob) [reply prospero-common/src/main/java/org/wildfly/prospero/actions/ApplyPatch.java:118]: Yes, good idea. I want to try and extract the whole section creating ProvisionManager and associated resources to utility at some point

**Comment by spyrkob** [comment]
> Would the patch repo remain in the .installation/ permanently?
Ideally the installed artifacts could be resolved from modules (#65) - then the patch repository can go away.

**Pull request merged**